nano/nano_node/entry.cpp

```diff
@@ -1355,29 +1356,81 @@ int main (int argc, char * const * argv)
 			auto handler_l (std::make_shared<nano::json_handler> (*inactive_node_l->node, config, command_l.str (), response_handler_l));
 			handler_l->process_request ();
 		}
-		else if (vm.count ("debug_validate_blocks"))
+		else if (vm.count ("validate_blocks") || vm.count ("debug_validate_blocks"))
 		{
+			nano::timer<std::chrono::seconds> timer;
+			timer.start ();
 			auto inactive_node = nano::default_inactive_node (data_path, vm);
 			auto node = inactive_node->node;
-			auto transaction (node->store.tx_begin_read ());
-			std::cout << boost::str (boost::format ("Performing blocks hash, signature, work validation...\n"));
-			size_t count (0);
-			uint64_t block_count (0);
-			for (auto i (node->store.latest_begin (transaction)), n (node->store.latest_end ()); i != n; ++i)
+			bool const silent (vm.count ("silent"));
+			unsigned threads_count (1);
+			auto threads_it = vm.find ("threads");
+			if (threads_it != vm.end ())
 			{
+				try
+				{
+					threads_count = boost::lexical_cast<unsigned> (threads_it->second.as<std::string> ());
+				}
+				catch (boost::bad_lexical_cast &)
+				{
+					std::cerr << "Invalid threads count\n";
+					result = -1;
```

Comment:
Should exit here, currently it's continuing with 1 thread.

Reply:
Might also consider the try convert method instead of the exception one (`boost::lexical_cast` just calls this under the hood and throws exception if it fails):
```
if (!boost::conversion::try_lexical_convert (threads_it->second.as<std::string> (), threads_count)) {
  std::cerr
}
```

Reply:
Should we also use default hardware_concurrency number of threads for this? It doesn't use the whole CPU on my system, still limited by disk even with NVME

Reply:
I don't think so. If use all threads, then you probably can't run daemon in background. With fast i/o it can use all CPU. I.e. with Optane 900P I have full usage of 18 threads (according to `top`). I guess same can apply to some slower cloud vCPU + NVME SSD.

Reply:
Fair enough, a callout on documentation instead 👍 
But if I provide `--threads abc` it's not stopping early at the moment.

Reply:
Updated